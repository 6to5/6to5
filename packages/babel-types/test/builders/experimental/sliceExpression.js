import * as t from "../../..";

describe("builders", function() {
  describe("experimental", function() {
    describe("sliceExpression", function() {
      it("should validate", function() {
        expect(
          t.sliceExpression(
            t.identifier("foo"),
            t.identifier("bar"),
            t.numericLiteral(42),
            t.unaryExpression("-", t.numericLiteral(42)),
          ),
        ).toMatchSnapshot();
        expect(
          t.sliceExpression(
            t.identifier("foo"),
            t.unaryExpression("-", t.identifier("Infinity")),
          ),
        ).toMatchSnapshot();
        expect(() =>
          t.sliceExpression(
            t.identifier("foo"),
            t.binaryExpression("+", t.numericLiteral(4), t.numericLiteral(2)),
          ),
        ).toThrow();
      });
    });
  });
});
