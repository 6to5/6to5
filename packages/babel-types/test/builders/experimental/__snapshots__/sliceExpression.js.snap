// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`builders experimental sliceExpression should validate 1`] = `
Object {
  "lower": Object {
    "name": "bar",
    "type": "Identifier",
  },
  "object": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "step": Object {
    "argument": Object {
      "type": "NumericLiteral",
      "value": 42,
    },
    "operator": "-",
    "prefix": true,
    "type": "UnaryExpression",
  },
  "type": "SliceExpression",
  "upper": Object {
    "type": "NumericLiteral",
    "value": 42,
  },
}
`;

exports[`builders experimental sliceExpression should validate 2`] = `
Object {
  "lower": Object {
    "argument": Object {
      "name": "Infinity",
      "type": "Identifier",
    },
    "operator": "-",
    "prefix": true,
    "type": "UnaryExpression",
  },
  "object": Object {
    "name": "foo",
    "type": "Identifier",
  },
  "step": null,
  "type": "SliceExpression",
  "upper": null,
}
`;
